plugins/modules/exo_rdns_record.py

```diff
@@ -0,0 +1,192 @@
+    state: absent
+'''
+
+RETURN = '''
+---
+exo_rdns_domain:
+  description: Reverse DSN name for the cs instance
+  returned: success
+  type: str
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_bytes, to_text
+from ansible_collections.ngine_io.cloudstack.plugins.module_utils.cloudstack import (
+    AnsibleCloudStack,
+    cs_argument_spec,
+    cs_required_together
+)
+
+
+class ExoRdnsRecord(AnsibleCloudStack):
+
+    def __init__(self, module):
+        super(ExoRdnsRecord, self).__init__(module)
+
+        self.name = self.module.params.get('name')
+        self.content = self.module.params.get('content')
+
+        self.returns = {
+            'domain': 'domain'
+        }
+
+        self.instance = None
+
+    def get_instance(self):
+        instance = self.instance
+        if not instance:
+            args = {
+                'account': self.get_account(key='name'),
+                'domainid': self.get_domain(key='id'),
+                'projectid': self.get_project(key='id'),
+                'fetch_list': True,
+            }
+            # Do not pass zoneid, as the instance name must be unique across
+            # zones.
+            instances = self.query_api('listVirtualMachines', **args)
+            if instances:
+                for v in instances:
+                    if self.name.lower() in [v['name'].lower(), v['displayname'].lower(), v['id']]:
+                        self.instance = v
+                        break
+
+        return self.instance
+
+    def get_record(self, instance):
+        result = {}
+
+        res = self.query_api('queryReverseDnsForVirtualMachine',
+                             id=instance['id'])
+        nics = res['virtualmachine'].get('nic', [])
+
+        defaultnics = [nic for nic in nics if nic.get('isdefault', False)]
+
+        if len(defaultnics) > 0:
+            domains = [record['domainname'] for record
+                       in defaultnics[0].get('reversedns', [])
+                       if 'domainname' in record]
+
+            if len(domains) > 0:
+                result = {
+                    'domainname': domains[0],
+                }
+
+        return result
+
+    def present_record(self):
+        instance = self.get_instance()
+
+        if not instance:
+            self.module.fail_json(
+                msg="No virtual machine with name=%s found. " % self.name)
+
+        data = {
+            'domainname': self.content,
+        }
+        record = self.get_record(instance)
+        if self.has_changed(data, record):
+            self.result['changed'] = True
+            self.result['diff']['before'] = record
+            self.result['diff']['after'] = data
+            if not self.module.check_mode:
+                self.query_api('updateReverseDnsForVirtualMachine',
+                               id=instance['id'],
+                               domainname=data['domainname'])
+
+        return data
```

Comment:
Rather then returning the user input, I would prefer to return what the api returns in the result of [updateReverseDnsForVirtualMachine](https://community.exoscale.com/api/compute/#updatereversednsforvirtualmachine_GET)

Reply:
Regrettably I can hardly help with that. I currently neither have access to exoscale nor do I have a use case for the return values. I'd rather prefer to leave that up to the maintainers / active users of this collection.